components/scream/docs/source-tree.md

```diff
@@ -0,0 +1,63 @@
+# SCREAM's Source Tree
+
+All SCREAM-specific code can be found in `components/scream` within the
+[SCREAM repo](https://github.com/E3SM-Project/scream). Here's how things are
+organized:
+
++ `bin`: Miscellaneous tools for helping with the build process. Currently, this
+  directory only contains a template for generating the `scream_mpicxx` compiler
+  script.
++ `cime_config`: Tools and XML files for integrating SCREAM with E3SM via the
+  CIME framework.
++ `cmake`: CMake functions and macros used by the configuration/build system.
++ `data`: Data files used by our tests.
++ `docs`: Documentation for the SCREAM project, including design documents,
+  instructions for building and testing SCREAM, and this document.
++ `extern`: Source for certain lightweight third-party libraries, embedded
+  directly into the repo (and not imported as submodules).
++ `scripts`: Miscellaneous scripts that implement workflows for running tests
+  and analyzing performance.
++ `src`: All C++ source code (and any bridges to Fortran) for SCREAM are stored
+  here. We describe the contents of this directory in greater detail below.
++ `tests`: Implements standalone, end-to-end tests for various SCREAM
+  components (RRTMG, HOMME, P3, SHOC, etc).
+
+In addition, you'll notice the following files in `components/scream`:
+
++ `CMakeLists.txt`: The CMake file that defines SCREAM's configuration/build
```

Comment:
Since CMakeLists.txt and tests/ is included in every directory in components/scream/ (or at least in src/ subdirectories), perhaps it would be worth moving mention of these to a special section about patterns found in all directories?

Reply:
I think we could actually avoid mentioning the CMakeLists.txt files. If one doesn't know cmake at all, we have a bigger problem than the documentation. Honestly, we could probably skip the description of the individual files, and focus only on the directories. That's what the user need to know, in order to find the code he/she needs (or in order to find out where he/she should put his/her code).

Reply:
Up to you. I just mentioned them to give the reader the impression that these things are around and that the build system isn't magical.

I don't think that many people "know" CMake, to be honest. And the CMake docs are pretty bad for beginners. :-)

Reply:
Fair point. Then we could simply say that files in the repo that start with `CMake` or `CTest` or have extension `.cmake` are files that contain cmake-syntax code, used to configure, build, and test the project. And maybe add a link to some cmake documentation (even if not great) for those who know little-to-nothing about cmake.

Reply:
Yeah, a lot of people who know nothing about CMake - or C++ - will end up cutting their teeth on our code base... so yeah, we have bigger problems, but every little bit helps.